src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -403,9 +404,39 @@ public String getQuestionResultStatisticsHtml(List<FeedbackResponseAttributes> r
             return "";
         }
 
-        boolean isContainsNonEmptyResponse = false; // we will only show stats if there is at least one nonempty response
-
         Map<String, Integer> answerFrequency = new LinkedHashMap<String, Integer>();
+        int numChoicesSelected = getQuestionResultStatistics(responses, answerFrequency);
+        if (numChoicesSelected == -1) {
+            return "";
+        } else {
+            DecimalFormat df = new DecimalFormat("#.##");
+
+            StringBuilder fragments = new StringBuilder();
+            for (Entry<String, Integer> entry : answerFrequency.entrySet()) {
+                fragments.append(Templates.populateTemplate(FormTemplates.MCQ_RESULT_STATS_OPTIONFRAGMENT,
+                        Slots.MCQ_CHOICE_VALUE, entry.getKey(),
+                        Slots.COUNT, entry.getValue().toString(),
+                        Slots.PERCENTAGE,
+                        df.format(100 * divideOrReturnZero(numChoicesSelected, (double) entry.getValue()))));
+            }
+            // Use same template as MCQ for now, until they need to be
+            // different.
+            return Templates.populateTemplate(FormTemplates.MCQ_RESULT_STATS, Slots.FRAGMENTS,
+                    fragments.toString());
+        }
+    }
+
+    private double divideOrReturnZero(int numChoice, double entryValue) {
```

Comment:
Consider switching the positions of the two parameters so that it is more obvious which value you are using as the denominator.
```
private double divideOrReturnZero(double entryValue, int numChoice) {
```
Perhaps accept entryValue as an int parameter and only cast it into a double in this method